src/EventEmitterTrait.php

```diff
@@ -55,49 +61,51 @@ public function once($event, callable $listener)
         return $this;
     }
 
-    public function removeListener($event, callable $listener)
+    public function removeListener(string $event, callable $listener): void
     {
-        if ($event === null) {
-            throw new InvalidArgumentException('event name must not be null');
+        if ($event === '') {
+            throw new InvalidArgumentException('event name must not be an empty string');
         }
 
         if (isset($this->listeners[$event])) {
             $index = array_search($listener, $this->listeners[$event], true);
+
             if (false !== $index) {
                 unset($this->listeners[$event][$index]);
-                if (count($this->listeners[$event]) === 0) {
+
+                if (\count($this->listeners[$event]) === 0) {
```

Comment:
Since we're on `8.1+`, why not use `use function count`?

Reply:
Simple enough to change. I suppose it could technically come down to preference. With compiler optimized functions, I normally do not import it with `use function`, opting to use `\...` instead.

Reply:
Added in latest commit.